src/NzbDrone.Api/Extensions/Pipelines/UrlBasePipeline.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using Nancy;
+using Nancy.Bootstrapper;
+using Nancy.Responses;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Configuration;
+
+namespace NzbDrone.Api.Extensions.Pipelines
+{
+    public class UrlBasePipeline : IRegisterNancyPipeline
+    {
+        private readonly string _urlBase;
+
+        public UrlBasePipeline(IConfigFileProvider configFileProvider)
+        {
+            _urlBase = configFileProvider.UrlBase;
+        }
+
+        public int Order => 99;
+
+        public void Register(IPipelines pipelines)
+        {
+            pipelines.BeforeRequest.AddItemToStartOfPipeline((Func<NancyContext, Response>) Handle);
+        }
+
+        private Response Handle(NancyContext context)
+        {
+
+            if (_urlBase.IsNotNullOrWhiteSpace() && _urlBase != context.Request.Url.BasePath)
```

Comment:
What if someone visits :8989/somethingelse/api/ then the redirect will be wrong :8989/base/somethingelse/api.
How about redirect for the root, and 404 for anything else?